### ISSUE TITLE
Broken docker compose output for SQL Server container logs

### ISSUE THREAD

**Issue by mloskot** [microsoft/mssql-docker#850]
Forwarding from https://forums.docker.com/t/broken-docker-compose-output-for-sql-server-container-logs/137128, has anyone experienced the issue with such **messy log stream output** while using `mcr.microsoft.com/mssql/server:{2017,2019,2022}-latest` with Docker Desktop 4.22.0 on Windows 11, regardless of terminal or Windows Terminal vs Windows Console host used:

- PowerShell 7
   ![1353116f34534e273227d15bfe3f6a900ec878cd](https://github.com/microsoft/mssql-docker/assets/80741/355ba54d-bb18-47e2-ba04-6d4b7cbacefd)
- Git Bash
   ![b2347baae67618de17814c23178ac69bfc3eb83b](https://github.com/microsoft/mssql-docker/assets/80741/21b229a5-de63-4945-a44e-682103a2caa3)
- Command Prompt (Windows Terminal)
   ![ef3e1140c462e82f7d9716f28fb9b9e7b260f9d3](https://github.com/microsoft/mssql-docker/assets/80741/f0ac9b42-6322-4ef7-84a7-674079b17ad6)
- Command Prompt (Windows Console Host)
   ![11a54472c1c9ba947b5f4af0a92d9c246838190c](https://github.com/microsoft/mssql-docker/assets/80741/d6baaed1-ebc8-4da6-9175-bb475c52f96e)

while after the `docker compose -d` and inspecting `docker compose logs` output:

![eef89d4513eb1a52b8f783f45b2dad87fe9d08e8](https://github.com/microsoft/mssql-docker/assets/80741/75aec2f7-0352-4962-8ebf-96297da8d91f)

**Comment by mloskot** [comment]
The thread https://forums.docker.com/t/broken-docker-compose-output-for-sql-server-container-logs/137128/6 received some very informative update and here is the gist quoted

> So some log messages actually start with `\r\t` and that’s when I see the compise output lines starting with an incomplete container name like “test-mss”. Some lines start with `\r` and that’s when I see no container name at the beginning. The perfectly working lines start with regular texts not `\r or `\r\t`.
> 
> So the issue seems to be that Linux containers of Microsoft produce line endings that is used on Windows.

**Comment by klondikemarlen** [comment]
I wonder how hard it would be to produce a log driver plugin that auto-converted the line endings? e.g. https://docs.docker.com/config/containers/logging/plugins/

**Comment by mloskot** [comment]
I wonder why Microsoft did not adjust the SQL Server behaviour on Linux, so it outputs correct endings. Anything else will eventually turn into broken prothesis. 

When in Rome, dance like Romans do.

**Comment by sbleon** [comment]
I would love to see this fixed.

**Comment by azwalzaman** [comment]
Facing the same issue

**Comment by davidezechukwu** [comment]
The latest version of SQL Server (mcr.microsoft.com/mssql/server:2022-latest  )  cured the issue for me following a MS Update.  The docker image on my machine was last updated on October 2023, so if the error is happening recently again, it may have bneen reintroduced, recently again, since then

**Comment by PureKrome** [comment]
@davidezechukwu Hi David - are you saying: "This issue has been fixed in the _latest_ (at this point of time) docker image" ?

this version / date:

![image](https://github.com/microsoft/mssql-docker/assets/899878/04d3ec21-af8e-44dc-b07c-5eee88a4aae8)

**Comment by davidezechukwu** [comment]
Yes. I am also using running Ubuntu 22.  My current sql server image is 2months old

**Comment by rsvoboda** [comment]
We have seen the same on RHEL 8.9 instances with podman in our Jenkins jobs. We didn't see this when running on RHEL 8.9 instances with docker.

Example of problematic output:
```
10:18:22.000 MSSQL:se
10:18:22.000 MSSQL:r
10:18:22.000 MSSQL:v
10:18:22.000 MSSQL:i
10:18:22.000 MSSQL:c
10:18:22.000 MSSQL:e
10:18:22.000 MSSQL:s
10:18:22.000 MSSQL:.
10:18:22.000 MSSQL:
10:18:22.000 MSSQL:
10:18:22.000 MSSQL:
10:18:23.000 MSSQL:2023
10:18:23.000 MSSQL:-12-19
10:18:23.000 MSSQL:10
```

Reproducer:
 - Have RHEL 8.9 instance with podman
 - get Quarkus code - `git clone https://github.com/quarkusio/quarkus`
 - compile it - `mvn clean install -Dquickly -Dno-test-modules`
 - run MSSQL tests - `mvn -f quarkus/integration-tests/pom.xml clean dependency:tree verify -V -B -fae -Denforcer.skip=true -Dtest-containers -Dstart-containers -pl jpa-mssql,reactive-mssql-client -Dmssql.image=mcr.microsoft.com/mssql/server:2022-latest`

It's Quarkus based as I work on Quarkus :)

**Comment by marksteward** [comment]
As a workaround you can use something like this in your docker-compose.yml:

```
    command: "bash -c \"/opt/mssql/bin/sqlservr | stdbuf -oL -eL tr -d '\r'\""
```

I haven't tested what happens with error lines, but this makes the standard output readable.
